### ISSUE TITLE
Can't make detach() work

### ISSUE THREAD

**Issue by vitorenesduarte** [orientechnologies/orientdb#3472]
```java
      ODatabaseDocument db = new ODatabaseDocumentTx("remote:localhost/Run");
      ODocument d;
      try {
         db.open("admin", "admin");
         d = new ODocument("User");
         d.field("name", "phoebe");
         d.save();
         d.detach();
         //d.deserializeFields(d.fieldNames());
      } finally {
         db.close();
      }
      System.out.println(d.toJSON());
```

This does not work. Although, if I comment the __d.detach();__ line, and uncomment the line below, it works.
What am I doing wrong?

**Comment by lvca** [comment]
Wich version? (Please report ALWAYS version to avoid such message)

**Comment by vitorenesduarte** [comment]
It was 2.0-rc1. I've updated, and it works.
Sorry.

**Comment by vitorenesduarte** [comment]
In this case, neither commented lines work.
```java
      ODatabaseDocument db = new ODatabaseDocumentTx("remote:localhost/Demo");
      List<ODocument> resultList;
      try {
         db.open("admin", "admin");
         resultList = db.command(new OSQLSynchQuery<ODocument>("select from User limit 10")).execute();
         for (ODocument d : resultList) {
            //d.deserializeFields(d.fieldNames());
            //d.detach();
         }
      } finally {
         db.close();
      }
      for (ODocument d : resultList) {
         System.out.println(d.toJSON());
      }
```

```bash
com.orientechnologies.orient.core.exception.ODatabaseException: Database instance is not set in current thread. Assure to set it with: ODatabaseRecordThreadLocal.INSTANCE.set(db);
	at com.orientechnologies.orient.core.db.ODatabaseRecordThreadLocal.get(ODatabaseRecordThreadLocal.java:52)
	at com.orientechnologies.orient.client.remote.OSBTreeBonsaiRemote.firstKey(OSBTreeBonsaiRemote.java:230)
	at com.orientechnologies.orient.core.db.record.ridbag.sbtree.OSBTreeRidBag$SBTreeMapEntryIterator.init(OSBTreeRidBag.java:483)
	at com.orientechnologies.orient.core.db.record.ridbag.sbtree.OSBTreeRidBag$SBTreeMapEntryIterator.<init>(OSBTreeRidBag.java:417)
	at com.orientechnologies.orient.core.db.record.ridbag.sbtree.OSBTreeRidBag.rawIterator(OSBTreeRidBag.java:524)
	at com.orientechnologies.orient.core.db.record.ridbag.ORidBag.rawIterator(ORidBag.java:152)
	at com.orientechnologies.orient.core.fetch.OFetchHelper.processRecordRidMap(OFetchHelper.java:143)
	at com.orientechnologies.orient.core.fetch.OFetchHelper.fetch(OFetchHelper.java:70)
	at com.orientechnologies.orient.core.serialization.serializer.record.string.ORecordSerializerJSON.toString(ORecordSerializerJSON.java:327)
	at com.orientechnologies.orient.core.serialization.serializer.record.string.ORecordSerializerStringAbstract.toString(ORecordSerializerStringAbstract.java:665)
	at com.orientechnologies.orient.core.record.ORecordAbstract.toJSON(ORecordAbstract.java:192)
	at com.orientechnologies.orient.core.record.ORecordAbstract.toJSON(ORecordAbstract.java:188)
	at ...
```


I assume it's because of the edges linked to these records. __detach()__ does not work in these situations?

[Here](https://gist.github.com/vitorenesduarte/f8e1c205768243bb24fe)'s an example of what a user looks like in my db.

**Comment by vitorenesduarte** [comment]
```java
      ODatabaseDocument db = new ODatabaseDocumentTx("remote:localhost/Demo");
      ODocument user;
      try {
         db.open("admin", "admin");
         List<ODocument> resultList = db.command(new OSQLSynchQuery<ODocument>("select from User limit 1")).execute();
         user = resultList.get(0);
         user.detach();
      } finally {
         db.close();
      }
      for (String s : user.fieldNames()) {
         System.out.println(s);
      }
```
This shows the fields were detached. But if any of those fields is an ORidBag, I can't do nothing with it.

**Comment by lvca** [comment]
Detaching a ridbag with remote connection could mean loading millions of items on the client side. Now detach() works only with plocal connections. You could rather avoiding using bonsai in ridbag by setting this option at client and server sides:

    -Dindex.embeddedToSbtreeBonsaiThreshold=-1

But huge edge collection will be much slower.

**Comment by vitorenesduarte** [comment]
I don't want to detach the records inside ORidBag. 
I would like to know the size of the ORidBag, the RID's inside, ... but it seems I can't.
Thanks @lvca :)

**Comment by lvca** [comment]
I think this is not supported out of the box, sorry.

**Comment by vitorenesduarte** [comment]
@lvca Is it possible to detach only one depth-level?

**Comment by lvca** [comment]
Nice feature to have :-) Please, could you open a new issue for that?

**Comment by vitorenesduarte** [comment]
#3482

Don't know if it's understandable.
